nixos/modules/services/system/nscd.nix

```diff
@@ -69,18 +92,23 @@ in
           config.environment.etc.hosts.source
           config.environment.etc."nsswitch.conf".source
           config.environment.etc."nscd.conf".source
+        ] ++ optionals config.users.mysql.enable [
+          config.environment.etc."libnss-mysql.cfg".source
+          config.environment.etc."libnss-mysql-root.cfg".source
         ];
 
-        # We use DynamicUser because in default configurations nscd doesn't
-        # create any files that need to survive restarts. However, in some
-        # configurations, nscd needs to be started as root; it will drop
-        # privileges after all the NSS modules have read their configuration
-        # files. So prefix the ExecStart command with "!" to prevent systemd
-        # from dropping privileges early. See ExecStart in systemd.service(5).
+        # In some configurations, nscd needs to be started as root; it will
+        # drop privileges after all the NSS modules have read their
+        # configuration files. So prefix the ExecStart command with "!" to
+        # prevent systemd from dropping privileges early. See ExecStart in
+        # systemd.service(5). We use a static user, because some NSS modules
+        # sill want to read their configuration files after the privilege drop
+        # and so users can set the owner of those files to the nscd user.
```

Comment:
Isn't it sufficient to work with `SupplementaryGroups` here ot make sure that whatever group owns the config files is also used by the service?

Anyways, cc @arianvp @flokli @mweinelt.

Reply:
Yes, that DynamicUser + SupplementaryGroups is the way to go.

Reply:
Sure that would be possible but I don't like this solution, because we still have to define the group which will later be passed to `SupplementaryGroups` and using another group than a `nscd` group wouldn't make sense because `libnss-mysql` is a part of `nscd`.

And when we define this group, its name should still be configurable and then in my opinion it will look strange if we have a `services.nscd.group` option and no `services.nscd.user` option.

The next thing is, that with `SupplementaryGroups`, the config files have to be group-readable, which is okay when the group has only one user in it, but I still prefer the solution with the static user and group.

Reply:
Swapping out DynamicUser for static user/group is not the same in terms of hardening. If you insist on removing DynamicUser read up on what other configuration options you're removing.

Reply:
Oh I didn't know that. Then I will change it to `SupplementaryGroups`. Thanks for the hint :)

Reply:
Okay I have tested it with `SupplementaryGroups` but for some reason, `libnss-mysql` refused to work. The config files were group-readable and I can read them with `systemd-run -p DynamicUser=true -p SupplementaryGroups=[mygroup] -t bash`, but for some reason, `libnss-mysql` does not work with that. It works when I set the mode of the config files to 0644, but obviously we don't want them to be world-readable.

So I have rolled back to the static user and group and applied the hardenings from `DynamicUser` manually. (I will push when I have finished the whole refactoring suggested in the other threads)

Reply:
Now I have pushed the hardenings.